3D/senior-csharp-test/Assets/Cache/FileCacheStorage.cs

```diff
@@ -1,22 +1,26 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
-using System.Text;
+using UnityEngine;
 
 [ Serializable ]
 public class FileCacheStorage : ICacheStorage
 {
-	private const string ACTUAL_VERSION = "1.0";
+	private const string CURRENT_VERSION = "1.0";
 	private const string DATE_FORMAT = "yyMMdd";
 	private const float EXPIRING_TIME_DAYS = 60f;
+
 	private readonly string _cachePath;
-	private readonly Dictionary<string, FileCacheEntry> _cacheEntries = new Dictionary<string, FileCacheEntry>();
+	private readonly Dictionary<string, FileCacheEntry> _cacheEntryById = new Dictionary<string, FileCacheEntry>();
 	private readonly string _fileName;
 	private readonly IFileSystem _fileSystem;
 	private readonly ISerializer _serializer;
-	public List<FileCacheEntry> CacheEntries = new List<FileCacheEntry>();
```

Comment:
you renamed this, so I'm not sure the cache deserializarion would work... or the json element would have a weird name... In any case we can't to the same rename in Display, it will break the existing cache 

maybe we should add a json property above. Or (I would find that even prettier), not have this list at all, but deserialize into a list directly and pass it to the method that fills in the dictionary

Reply:
I've tried to explain in the MR description at the top what I've done here, I went the quick way for this MR but I can change and use Json.Net yes and use Property (don't have that in JsonUtility). I I can try also what you propose (use JsonUtility.FromJson()), What do you prefer ?

Reply:
Finally I used Json.net (better to be able to deserialize to a list of interfaces so that the candidate can test) and I'm deserializing in a new list directly that I pass to the method that fills the dictionary from it